docs/execution_providers/ACL-ExecutionProvider.md

```diff
@@ -0,0 +1,54 @@
+## ACL Execution Provider
+
+[Arm Compute Library](https://github.com/ARM-software/ComputeLibrary) is an open source inference engine maintained by Arm and Linaro companies. The integration of ACL as an execution provider (EP) into ONNX Runtime accelerates performance of ONNX model workloads across Armv8 cores.
+
+### Build ACL execution provider
+Developers can use ACL library through ONNX Runtime to accelerate inference performance of ONNX models. Instructions for building the ACL execution provider from the source is available below.
+
+### Supported BSP
+* i.MX8QM BSP
+Install i.MX8QM BSP:
+```
+source fsl-imx-xwayland-glibc-x86_64-fsl-image-qt5-aarch64-toolchain-4*.sh
+```
+
+Setup build environment:
+```
+source /opt/fsl-imx-xwayland/4.*/environment-setup-aarch64-poky-linux
+alias cmake="/usr/bin/cmake -DCMAKE_TOOLCHAIN_FILE=$OECORE_NATIVE_SYSROOT/usr/share/cmake/OEToolchainConfig.cmake"
+```
+
+Confiure ONNX Runtime with ACL support:
+```
+cmake ../onnxruntime-arm-upstream/cmake -DONNX_CUSTOM_PROTOC_EXECUTABLE=/usr/bin/protoc -Donnxruntime_RUN_ONNX_TESTS=OFF -Donnxruntime_GENERATE_TEST_REPORTS=ON -Donnxruntime_DEV_MODE=ON -DPYTHON_EXECUTABLE=/usr/bin/python3 -Donnxruntime_USE_CUDA=OFF -Donnxruntime_USE_NSYNC=OFF -Donnxruntime_CUDNN_HOME= -Donnxruntime_USE_JEMALLOC=OFF -Donnxruntime_ENABLE_PYTHON=OFF -Donnxruntime_BUILD_CSHARP=OFF -Donnxruntime_BUILD_SHARED_LIB=ON -Donnxruntime_USE_EIGEN_FOR_BLAS=ON -Donnxruntime_USE_OPENBLAS=OFF -Donnxruntime_USE_ACL=ON -Donnxruntime_USE_MKLDNN=OFF -Donnxruntime_USE_MKLML=OFF -Donnxruntime_USE_OPENMP=ON -Donnxruntime_USE_TVM=OFF -Donnxruntime_USE_LLVM=OFF -Donnxruntime_ENABLE_MICROSOFT_INTERNAL=OFF -Donnxruntime_USE_BRAINSLICE=OFF -Donnxruntime_USE_NUPHAR=OFF -Donnxruntime_USE_EIGEN_THREADPOOL=OFF -Donnxruntime_BUILD_UNIT_TESTS=ON -DCMAKE_BUILD_TYPE=RelWithDebInfo
```

Comment:
see other examples of building ONNX Runtime using build.sh which is more convenient.
ideally the build should be as simple as ./build.sh --config RelWithDebInfo --use_acl

Reply:
We have stuck to ONNX Runtime ARM cross compilation build steps as documented here:
https://github.com/microsoft/onnxruntime/blob/master/BUILD.md#cross-compiling-on-linux-without-docker
For the moment we chose to call directly CMake due to its similarity with Yocto BSP integration and the fact that ACL external dependency is delivered pre-built with the cross-compilation tool-chain.

Reply:
Are you working on Yocto ?

---
In reply to: [339558855](https://github.com/microsoft/onnxruntime/pull/2258#discussion_r339558855) [](ancestors = 339558855)

Reply:
Yes, both the BSP and the cross-compiler are built with Yocto.

Reply:
build.sh can be addressed after ACL EP merge, it is not prepared yet for cross-compilation.